ros/src/twist_controller/twist_controller.py

```diff
@@ -6,12 +11,43 @@
 
 class Controller(object):
 
-    def __init__(self, wheel_base, steer_ratio, min_speed, max_lat_accel, max_steer_angle):
+    def __init__(self, wheel_base, steer_ratio, min_speed, max_lat_accel, max_steer_angle, max_decceleration, max_acceleration, fuel_capacity, vehicle_mass, wheel_radius):
+        self.fuel_capacity = fuel_capacity
+        self.vehicle_mass = vehicle_mass
+        self.wheel_radius = wheel_radius
+        self.current_timestep = None
+
         self.yaw_controller = YawController(
             wheel_base, steer_ratio, min_speed, max_lat_accel, max_steer_angle)
 
+        self.velocity_pid_controller = PID(
+            0.1, 0.000005, 0.05, max_decceleration, max_acceleration)
+
     def control(self, target_linear_velocity, target_angular_velocity, current_linear_velocity):
-        # TODO - implement proper throttle and brake measurements
+        # use velocity pid controller to compute new velocity
+        acceleration = self.compute_acceleration(
+            target_linear_velocity - current_linear_velocity)
+        rospy.logdebug('Current linear velocity %s, target linear velocity %s, computed acceleration %s',
+                       current_linear_velocity, target_linear_velocity, acceleration)
 
         # Return throttle, brake, steer
-        return 0., 0., self.yaw_controller.get_steering(target_linear_velocity, target_angular_velocity, current_linear_velocity)
+        return (acceleration if acceleration > 0. else 0.,
+                self.compute_brake_torque(-acceleration) if acceleration < 0. else 0.,
```

Comment:
I got a little confused by the sign flips.  Maybe abs(acceleration) would be more clear?

Reply:
Sure. Feel free to comment anytime something isn't quite clear; the code needs to be maintainable by everyone. I'll adjust.